web/src/app.rs

```diff
@@ -118,6 +129,106 @@ fn SearchBar(cx: Scope, search: RwSignal<String>) -> impl IntoView {
     }
 }
 
+async fn update_metadata(update_metadata_request: &UpdateMetadataRequest) -> String {
+    let client = reqwest::Client::new();
+    match client
+        .post("http://localhost:3000/metadata")
+        .json(update_metadata_request)
+        .send()
+        .await
+    // Below handling are useless for now.
+    {
+        Ok(response) => {
+            println!("response: {:?}", response);
+            "yes!".to_string()
+        }
+        Err(e) => {
+            println!("error: {:?}", e);
+            "nope".to_string()
+        }
+    }
+}
+
+#[component]
+fn FileDetails(cx: Scope, full_file_name: String, metadata: Option<Metadata>) -> impl IntoView {
+    let (_metadata_updater, set_metadata_updater) = create_signal(
+        cx,
+        create_resource(cx, || (), |_| async move { "".to_string() }),
+    );
+
+    // we'll use a NodeRefs to store references to the input elements
+    // these will be filled when the elements are created
+    let note_element: NodeRef<Input> = create_node_ref(cx);
+    let priority_element: NodeRef<Select> = create_node_ref(cx);
+    let reviewer_element: NodeRef<Select> = create_node_ref(cx);
```

Comment:
Interesting, have not felt the need to use NodeRefs in a very long time and I don't see why you need them here.